client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -109,14 +109,29 @@ public class TrinoPreparedStatement
     private final String statementName;
     private final String originalSql;
     private boolean isBatch;
+    private boolean prepareStatementExecuted;
 
     TrinoPreparedStatement(TrinoConnection connection, Consumer<TrinoStatement> onClose, String statementName, String sql)
             throws SQLException
     {
         super(connection, onClose);
         this.statementName = requireNonNull(statementName, "statementName is null");
         this.originalSql = requireNonNull(sql, "sql is null");
-        super.execute(format("PREPARE %s FROM %s", statementName, sql));
+        prepareStatementIfNecessary(false);
```

Comment:
I feel that by trying to use `prepareStatementIfNecessary()` here (instead of conditionally preparing the statement) makes the implementation less readable and the intent less clear.  But that's just my opinion.

Reply:
Fair enough. Updated.